dpe/src/dpe_instance.rs

```diff
@@ -151,27 +152,27 @@ impl DpeInstance {
         handle: &ContextHandle,
         locality: u32,
     ) -> Result<usize, DpeErrorCode> {
-        let mut valid_handles = self
+        let mut valid_localities = self
             .contexts
             .iter()
             .enumerate()
             .filter(|(_, context)| {
-                context.state == ContextState::Active && &context.handle == handle
+                context.state == ContextState::Active && context.locality == locality
```

Comment:
I think it should be ok to just filter on `context.state == ContextState::Active && context.locality == locality && constant_time_eq(&context.handle.0, &handle.0))` right? And then if `valid_handles` size is != 1, we can return an error that the context state is corrupted.

Reply:
The reason I initially split this into two iterators was to differentiate the errors better. Note how we return InvalidLocality/ InvalidHandle based on which filter renders an empty iterator.

Reply:
Ok, that's a good reason. In that case, I would just suggest adding comments to each block explaining what checks are being done in each part and why they're separated.

Reply:
Done